### PR TITLE
chore(core): makes boot sequence clearer

### PR DESCRIPTION
**Pull request by mrclay**
**Makes boot sequence clearer**

Removes the unnecessary [boot, system] event. The installer was only using it to unregister a function that no longer exists.

Initializes context service directory after instantiation.

### CODE REVIEW

File: engine/classes/Elgg/Http/Input.php [hunk 1]
```diff
@@ -12,20 +12,13 @@
...
+	 * @var array
 	 */
-	private $CONFIG;
-
-	/**
-	 * Constructor
-	 */
-	public function __construct() {
-		global $CONFIG;
-		$this->CONFIG = $CONFIG;
```
Comment (mrclay) [inline engine/classes/Elgg/Http/Input.php]: - [ ] need BREAKING notice. Don't depend on $CONFIG->input being set.

File: engine/lib/navigation.php [hunk 2]
```diff
@@ -126,7 +130,7 @@ function elgg_register_menu_item($menu_name, $menu_item) {
  * @since 1.8.0
  */
 function elgg_unregister_menu_item($menu_name, $item_name) {
-	global $CONFIG;
+	$CONFIG = _elgg_services()->config->getStorageObject();
```
Comment (jeabakker) [inline engine/lib/navigation.php]: why not use `$menus = elgg_get_config('menus');` like you do in the other functions in this file?

File: engine/lib/navigation.php [hunk 3]
```diff
@@ -154,7 +158,7 @@ function elgg_unregister_menu_item($menu_name, $item_name) {
  * @since 1.8.0
  */
 function elgg_is_menu_item_registered($menu_name, $item_name) {
-	global $CONFIG;
+	$CONFIG = _elgg_services()->config->getStorageObject();
```
Comment (jeabakker) [inline engine/lib/navigation.php]: why not use `$menus = elgg_get_config('menus');` like you do in the other functions in this file?

File: mod/uservalidationbyemail/start.php [hunk 4]
```diff
@@ -68,12 +68,12 @@ function uservalidationbyemail_disable_new_user($hook, $type, $value, $params) {
 	// another plugin is requesting that registration be terminated
 	// no need for uservalidationbyemail
 	if (!$value) {
-		return $value;
+		return;
```
Comment (jeabakker) [inline mod/uservalidationbyemail/start.php]: don't forget to change the returnvalue of the function in the function docs

**Review by jeabakker — CHANGES_REQUESTED** [review]

**Comment by mrclay** [comment]
Decided to simplify this into multiple PRs.

**Pull request merged**